### PR TITLE
Running ingestion in batch

### PR DESCRIPTION
**Pull request by vegetable68**
- Successfully finished running on a short list of 10 files
    - Took approx half a day
    - Output in wikidetox_conversation.ingested_conversations_short_10
-  Started running ingestion on the full list(excluding the 10 dumps) in batch of 20
    - Reasoning for running in batch: result is written to bigquery table iff the job is successfully finished, in order to prevent unexpected errors stopping successfully processed results being stored.
-  Current speed estimate: 24 batches will be finished in 2 weeks.
-  Will proceed coding up reconstruction on the ingested short list.

### CODE REVIEW

File: conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py [hunk 1]
```diff
@@ -78,27 +75,21 @@ def run(argv = None):
...
     '--staging_location=gs://wikidetox-viz-dataflow/staging',
     '--temp_location=gs://wikidetox-viz-dataflow/tmp',
-    '--job_name=yiqing-ingest-job-truncated-content',
+    '--job_name=yiqing-ingest-job-truncated-content-run-in-batch',
     '--num_workers=90',
   ])
 
   pipeline_options = PipelineOptions(pipeline_args)
   pipeline_options.view_as(SetupOptions).save_main_session = True
+  if known_args.batchno:
```
Comment (iislucas) [inline conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py]: I don't see where the value of `known_args` is set... ?

File: conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py [hunk 2]
```diff
@@ -32,6 +32,7 @@
 import json
 import sys
 import zlib
+import copy
```
[the remark this answers is not in the document]
Comment (vegetable68) [reply conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py]: copy was used in function truncate_content to divide one record into two.

File: conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py (hunk 1 above)
Comment (vegetable68) [reply conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py]: It's in line 72, I think in this commit that line was folded.

File: conversation_reconstruction_dataflow/ingest_revisions/get_input_list.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+full_set = []
```
[the remark this answers is not in the document]
Comment (vegetable68) [reply conversation_reconstruction_dataflow/ingest_revisions/get_input_list.py]: ah this is a temporary file, removed it, but added the other one in the new commit

File: conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+folder = 'input_lists/'
```
[the remark this answers is not in the document]
Comment (vegetable68) [reply conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py]: Pushed a new commit with comment of this and license staff.

File: conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py (hunk 1 above)
Comment (iislucas) [reply conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py]: Oops, so it is. Nice.

File: conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iislucas) [reply conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py]: ah, also in fold. cool.

File: conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (iislucas) [reply conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py]: Ah, cool. In that case I don't think we need to batched files to be uploaded as they are generated by this script, right?

[the remark this answers is not in the document]
Comment (iislucas) [reply conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py]: And we should include in the README some instructions to run this? or have it as part of setup?

[the remark this answers is not in the document]
Comment (vegetable68) [reply conversation_reconstruction_dataflow/ingest_revisions/truncate_input_list.py]: Yep batched files removed. Also changed the README content, it was a bit outdated.

**Comment by iislucas** [comment]
Nice! Feel free to merge on.

**Pull request merged**